lib/ace/edit_session/bracket_match.js

```diff
@@ -37,17 +37,76 @@ var Range = require("../range").Range;
 
 function BracketMatch() {
 
+
+    this.findSurroundingBrackets = function(position) {
+        if ((position.column == 0) && (position.row)) return null;
+
+        var allBrackets = { 
+            left : [this.$findOpeningBracket("}", position, /(\.?.paren)+/), 
+                    this.$findOpeningBracket(']', position, /(\.?.paren)+/), 
+                    this.$findOpeningBracket(')', position, /(\.?.paren)+/)],
+            right : [this.$findClosingBracket("{", position, /(\.?.paren)+/),
+                     this.$findClosingBracket('[', position, /(\.?.paren)+/),
+                     this.$findClosingBracket('(', position, /(\.?.paren)+/)]
+        };
+
+        var left = {row: null, column: null};
+        var right = {row: null, column: null};
+
+        var leftRowArr = [],
+            leftColArr = [],
+            rightColArr = [],
+            rightRowArr = [];
+
+        var length = Math.max(allBrackets.left.length, allBrackets.right.length)
+
+        for (var key = 0; key < length; key++) {
+            if (allBrackets.left[key] && allBrackets.right[key]) { 
```

Comment:
I think that in this case we'll want to return error, so that user could see that brackets in this piece of code do not match. The error should be meaningful, that is, we'll want to highlight the mismatch.